core/modules/parser/SelectParser.cc

```diff
@@ -237,6 +237,50 @@ class Antlr2Parser : public AntlrParser {
 };
 
 
+class Antlr4ErrorStrategy : public antlr4::DefaultErrorStrategy {
+public:
+    Antlr4ErrorStrategy(std::string const& statement) : _statement(statement) {}
+
+private:
+    void recover(antlr4::Parser *recognizer, std::exception_ptr e) override {
```

Comment:
Names of the private methods are supposed to begin with underscore (`void _recover()`) ...unless you're overriding a virtual method which isn't under your control (comes from ANTLR4)

Reply:
yes, it's an override of an antlr4 method :-(

Reply:
(and the same for the other ones below)